trollflow2/launcher.py

```diff
@@ -204,6 +204,22 @@ def message_to_jobs(msg, product_list):
     return jobs
 
 
+def _extract_filenames(msg):
+    """Extract the filenames from *msg*.
+
+    If the message contains a `filesystem` item, use fsspec to decode it.
+    """
+    filenames = [urlparse(uri).path for uri in gen_dict_extract(msg.data, 'uri')]
+    filesystems = list(gen_dict_extract(msg.data, 'filesystem'))
+    if filesystems:
+        from satpy.readers import FSFile
+        from fsspec.spec import AbstractFileSystem
+        import json
+        filenames = [FSFile(filename, AbstractFileSystem.from_json(json.dumps(filesystem)))
+                     for filename, filesystem in zip(filenames, filesystems)]
```

Comment:
Could this be a separate function for cleanliness?

Reply:
Good idea, thanks